src/modules/commander/Commander.cpp

```diff
@@ -2234,55 +2260,35 @@ Commander::run()
 
 			_last_manual_control_switches_arm_switch = _manual_control_switches.arm_switch;
 
-			if (arming_ret == TRANSITION_DENIED) {
-				/*
-				 * the arming transition can be denied to a number of reasons:
-				 *  - pre-flight check failed (sensors not ok or not calibrated)
-				 *  - safety not disabled
-				 *  - system not in manual mode
-				 */
-				tune_negative(true);
-			}
-
 			if (_manual_control_switches_sub.update(&_manual_control_switches) || safety_updated) {
 
 				// handle landing gear switch if configured and in a manual mode
-				if ((_manual_control_switches.gear_switch != manual_control_switches_s::SWITCH_POS_NONE) &&
+				if ((_vehicle_control_mode.flag_control_manual_enabled) &&
```

Comment:
OFFBOARD is a weird one I'll think about.

I still don't understand why ORBIT is a special thing that's neither AUTO or MANUAL. It could either be split as effectively 2 modes (depending on if you're using it from sticks or from mavlink) or as a single mode that's either AUTO or MANUAL at any given time. Then we cleanly know if it requires manual control or not, and hence how it ties into the failsafes.

Reply:
> It could either be split as effectively 2 modes

I'd not do that since other auto modes e.g. nudging during landing will have a similar hybrid situation and separate them doesn't scale.

> or as a single mode that's either AUTO or MANUAL at any given time

How about a single mode that's autonomous and RC is not required but input can be used if available. The same would fit for others e.g. nudging.

Reply:
I don't think this landing gear use case is an issue, the autonomous handling should be in the mode implementations and not in commander.